### PR TITLE
Update for latest ruby-lsp changes

### PR DESCRIPTION
**Pull request by andyw8**
Needed due to https://github.com/Shopify/ruby-lsp/pull/1648

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -20,8 +20,44 @@
...
+    # TODO: share with ruby-lsp?
+    sig do
+      type_parameters(:T)
+        .params(
+          source: T.nilable(String),
+          uri: URI::Generic,
+          block: T.proc.params(server: RubyLsp::Server, uri: URI::Generic).returns(T.type_parameter(:T)),
+        ).returns(T.type_parameter(:T))
+    end
+    def with_server(source = nil, uri = URI("file:///fake.rb"), &block)
```
[the remark this answers is not in the document]
Comment (vinistock) [reply test/test_helper.rb]: You'll need to `capture_subprocess_io` wherever we are executing the `initialized` notification.

File: lib/ruby_lsp/ruby_lsp_rails/addon.rb [hunk 2]
```diff
@@ -53,11 +54,11 @@ def create_code_lens_listener(response_builder, uri, dispatcher)
...
           response_builder: ResponseBuilders::Hover,
           nesting: T::Array[String],
-          index: RubyIndexer::Index,
           dispatcher: Prism::Dispatcher,
         ).void
       end
-      def create_hover_listener(response_builder, nesting, index, dispatcher)
+      def create_hover_listener(response_builder, nesting, dispatcher)
+        index = T.must(@global_state).index
         Hover.new(@client, response_builder, nesting, index, dispatcher)
```
[the remark this answers is not in the document]
Comment (andyw8) [reply lib/ruby_lsp/ruby_lsp_rails/addon.rb]: 👍 updated

File: lib/ruby_lsp/ruby_lsp_rails/hover.rb [hunk 3]
```diff
@@ -25,21 +25,21 @@ class Hover
...
           dispatcher: Prism::Dispatcher,
         ).void
       end
-      def initialize(client, response_builder, nesting, index, dispatcher)
+      def initialize(client, response_builder, nesting, global_state, dispatcher)
         @client = client
         @response_builder = response_builder
         @nesting = nesting
-        @index = index
+        @global_state = global_state
```
Comment (vinistock) [inline lib/ruby_lsp/ruby_lsp_rails/hover.rb:36]: Let's put the index in a variable to avoid doing `@global_state.index` multiple times.

**Review by vinistock — APPROVED** [review]
I put up a PR to ensure we deactivate addons on `with_server` https://github.com/Shopify/ruby-lsp/pull/1872. Otherwise, we spawn a bunch of runner clients and don't shut them down gracefully.

I think we can address the stderr printing in a separate PR, given that it needs to be addressed in the base LSP too.

**Pull request merged**